Sources/Sentry/SentryClient.m

```diff
@@ -238,6 +241,22 @@ - (SentryId *)sendEvent:(SentryEvent *)event
                   isCrashEvent:NO];
 }
 
+- (SentryTraceState *)getTraceStateWithEvent:(SentryEvent *)event withScope:(SentryScope *)scope
+{
+    id<SentrySpan> span;
+    if ([event isKindOfClass:[SentryTransaction class]]) {
+        span = [(SentryTransaction *)event trace];
+    } else {
+        span = scope.span;
+    }
```

Comment:
In the first meeting we had about this feature, I remember bruno telling us that the tracestate should be send if the scope has a transaction or if the event is a transaction.
So, i guess, in the case we are sending a transaction event, the trace in the envelope header should be the one from the event itself, because It is possible to create a transaction while having other in the scope.

@bruno-garcia Did I guess it right?

Reply:
I was not part of the meeting nor the design of the feature (I read the docs, but it didn't mention such details), @bruno-garcia reviewed the Java PR, I just checked a few specific things because of Android compatibility, so I don't have a formed opinion, I'll defer to @bruno-garcia

Reply:
> in the case we are sending a transaction event, the trace in the envelope header should be the one from the event itself, because It is possible to create a transaction while having other in the scope.

That is correct. Intuitively, you'll want the event payload and the trace state corresponding to each other, so that the transaction is sampled correctly.

Reply:
@maciejwalkowiak we probably have to change this then?

Reply:
@philipphofmann In Java:

- when transaction is captured - we get `traceState` from the transaction itself
- when event is captured - we get `traceState` from the scope, if there is an active transaction on the scope

As far as I understand, it's all good on the Java side ..?

Reply:
After a call with @jan-auer, I updated the wording for the dev docs in this PR a bit: https://github.com/getsentry/develop/pull/428. TLDR; the implementation here is correct to my understanding, and we need to change something on Java.

Reply:
Oh no, sorry, we don't need to change something in Java. I was mistaken; the implementation looks fine @marandaneto.